### PR TITLE
packaging:  Don't always build the kata-agent

### PR DESCRIPTION
**Pull request by fidencio**
There's absolutely no need to always build the Kata Containers agent when building the rootfs. We already have the agent being built / packaged on every PR, and we can rely on that to simply be unpacked into the rootfs, unless there's a change in the agent.

Please, see each commit for more info.

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks for the update - I hope this will save a lots of CI time and money!

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 1]
```diff
@@ -156,6 +156,24 @@ install_cached_tarball_component() {
 	mv "${component_tarball_name}" "${component_tarball_path}"
 }
 
+get_agent_tarball_path()
+{
```
[the remark this answers is not in the document]
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:160]: I will add it back :-)

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 2]
```diff
@@ -693,6 +692,8 @@ EOF
 		tar xvJpf ${AGENT_TARBALL} -C ${ROOTFS_DIR}
 	fi
 
+	${stripping_tool} ${ROOTFS_DIR}/usr/bin/kata-agent
```
Comment (danmihai1) [inline tools/osbuilder/rootfs-builder/rootfs.sh:695]: We can do this later, but perhaps we should have a way to bypass this new behavior.

**Comment by danmihai1** [comment]
A lot of goodness here - thanks @fidencio !

**Review by danmihai1 — APPROVED** [review]

File: tools/osbuilder/rootfs-builder/rootfs.sh (hunk 2 above)
Comment (fidencio) [reply tools/osbuilder/rootfs-builder/rootfs.sh:695]: I'd prefer doing this later, but just to confirm.

The idea here is that we can bypass the behaviour and actually **not** have the component installed, correct?
If so, I think we can enforce this for the attestation-agent / confiential-data-hub / opa, are those are extra components.  The kata agent itself must always be there, though.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

File: tools/osbuilder/rootfs-builder/rootfs.sh (hunk 2 above)
Comment (danmihai1) [reply tools/osbuilder/rootfs-builder/rootfs.sh:695]: Never mind. Yesterday I thought this rootfs.sh change was a significant change in symbol strip behavior - but I see now that it's consistent with the previous behavior.

**Pull request merged**